sdk/canton/community/base/src/main/scala/com/digitalasset/canton/data/ViewParticipantData.scala

```diff
@@ -223,7 +223,8 @@ final case class ViewParticipantData private (
         }
         RootAction(cmd, actors, failed, packagePreference)
 
-      case FetchActionDescription(inputContractId, actors, byKey, _version, templateIdO) =>
+      // TODO Should interfaceId be used? should it be passed to the command?
```

Comment:
If `Engine.reinterpret` can reinterpret a fetch replay command without knowing the interfaceId, you can leave it as is. Otherwise, you need to pass it to the command.

Wondering why you have added the interfaceId to the wire format in the first place?

Reply:
Right okay, it isn't needed to run reinterpret, we just need to be sure that the engine comes to the same conclusion. (i.e. it'll be compared later as part of view comparison)

Reply:
After discussion with remy, it is needed in replay and has been added.